crates/oxc_linter/src/rules/eslint/default_case.rs

```diff
@@ -0,0 +1,277 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+use regex::RegexBuilder;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn default_case_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("eslint(default-case): Require default cases in switch statements.")
+        .with_help("Add a default case.")
+        .with_labels([span0.into()])
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct DefaultCase(Box<DefaultCaseConfig>);
+
+#[derive(Debug, Clone)]
+pub struct DefaultCaseConfig {
+    comment_pattern: Regex,
+}
+
+impl Default for DefaultCaseConfig {
+    fn default() -> Self {
+        Self {
+            comment_pattern: RegexBuilder::new(r"^no default$")
```

Comment:
If the default is a simple string match, can we make this a fast path by defining

`comment_pattern: Option<Regex>`,

and just check for `no default` when it's `None`?

Constructing the regex and running it both introduce a small performance overhead.

Reply:
Ah, that makes sense! I'm still learning Rust :)

I'm not sure if I pushed the most Rust idiomatic approach but the tests are happy.